### DISCUSSION TITLE
Graph spawning multiple ops

### DISCUSSION THREAD

**Discussion by EtienneT** [dagster-io/dagster#14855] [Q&A]
I have a job downloading multiple zip files (one per month).  Then I extract the zip files and each file in the zip file is a xls I have to load as a dataframe.

I structured this job like so:

```
materialize_price_archive_job = materialize_price_archive.to_job()

@graph
def materialize_price_archive():
    allDates = get_dates()
    allDates.map(process_date)

@op(out=DynamicOut())
def get_dates():
    dates = pd.date_range('2017-01-01', '2019-12-01', freq='MS')
    for date in dates:
        yield DynamicOutput(date, mapping_key=f'date_{date:%Y%m%d}')

@graph
def process_date(date):
    basePathZip, basePathXls = get_folders()
    zipFilename = download_zip(date, basePathZip)

    allXls = extractZip(zipFilename, basePathXls)
    
    allXls.map(parseXls)

@op(out=DynamicOut())
def extractZip(context, zipFilename: Path, basePathXls: Path) -> List[DynamicOutput[Path]]:
    with zipfile.ZipFile(zipFilename, 'r') as zip_ref:
        allXls = zip_ref.infolist()
        zip_ref.extractall(basePathXls)

    # I have also tried just using yield here, but doesnt change anything
    return [DynamicOutput(basePathXls / file.filename, mapping_key=f'{Path(file.filename).stem}') for file in allXls]

@op
def parseXls(xlsPath):
  ...
```

parseXls is never executed and I get the following message in the log:
`Skipping step process_date.parseXls[date_20170101] due to skipped dependencies: ['process_date.extractZip[date_20170101].result'].`

Basically I want one op to download and extract my zip file.  Then I want this to create one op for each extracted file.

It seems like allXls.map(parseXls) doesnt work.  Not too sure if this is because I am in a nested graph. materialize_price_archive is a graph because it process multiple dates and process_date is a graph because it process multiple files per zip...
